test/tests/functional/pbs_resc_custom_perm.py

```diff
@@ -0,0 +1,164 @@
+        for rsc in self.rsc_list:
+            try:
+                self.server.manager(MGR_CMD_DELETE, RSC,
+                                    id=rsc, runas=ROOT_USER)
+            except PbsManagerError:
+                pass
+
+        self.momA = self.moms.values()[0]
+        self.momA.restart()
+        self.momA.delete_vnode_defs()
+        self.hostA = self.momA.shortname
+
+        rc = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+        self.assertEqual(rc, 0)
+        self.server.expect(NODE, {'state': 'free'}, id=self.hostA)
+
+        # Next set custom resources with m flag and without m flag
+        # via qmgr -c 'create resource'
+        attr = {'type': 'string', 'flag': 'h'}
+        r = 'foo_str'
+        rc = self.server.manager(
+            MGR_CMD_CREATE, RSC, attr, id=r, runas=ROOT_USER, logerr=False)
+        self.assertEqual(rc, 0)
+
+        attr = {'type': 'string', 'flag': 'hm'}
+        r = 'foo_strm'
+        rc = self.server.manager(
+            MGR_CMD_CREATE, RSC, attr, id=r, runas=ROOT_USER, logerr=False)
+        self.assertEqual(rc, 0)
+
+        # Create a custom resources via exechost_startup hook.
+        startup_hook_body = """
+import pbs
+e=pbs.event()
+localnode=pbs.get_local_nodename()
+e.vnode_list[localnode].resources_available['foo_strh'] = "str_hook"
+"""
+        hook_name = "start"
+        a = {'event': "exechost_startup", 'enabled': 'True'}
+        rv = self.server.create_import_hook(
+            hook_name,
+            a,
+            startup_hook_body,
+            overwrite=True)
+        self.assertTrue(rv)
+
+        self.momA.restart()
+        # Give the moms a chance to receive the updated resource.
+        # Ensure the new resource is seen by all moms.
+        m = self.momA.log_match("resourcedef;copy hook-related file",
+                                max_attempts=20, interval=1)
+        self.assertTrue(m)
+
+        # Create a exechost_begin hook.
+        begin_hook_body = """
+import pbs
+e=pbs.event()
+res_list = getattr(e.job, 'Resource_List')
+pbs.logjobmsg(e.job.id, "Resource_List is %s" % str(res_list))
+"""
+        hook_name = "begin"
+        a = {'event': "execjob_begin", 'enabled': 'True'}
+        rv = self.server.create_import_hook(
+            hook_name,
+            a,
+            begin_hook_body,
+            overwrite=True)
+        self.assertTrue(rv)
+
+    def test_custom_resc_single_node(self):
+        """
+        Test permission flag of resources of a single node job
+        using a execjob_begin hook.
+        """
+        self.logger.info("test_custom_resc__single_node")
+
+        a = {'Resource_List.foo_str': 'str_noperm',
+             'Resource_List.foo_strm': 'str_perm',
+             'Resource_List.foo_strh': 'str_hook'
+             }
+        j = Job(TEST_USER)
+        j.set_attributes(a)
```

Comment:
Consider making this more concise by setting the attributes in the Job() call.

Reply:
Done.